internal/app/buildclient/build.go

```diff
@@ -11,18 +11,43 @@ import (
 	"errors"
 	"fmt"
 	"os"
+	"strings"
 
 	build "github.com/sylabs/scs-build-client/client"
 )
 
 // buildArtifact sends a build request for the specified def and arch, optionally publishing it to
 // libraryRef. Output is streamed to standard output. If the build cannot be submitted, or does not
 // succeed, an error is returned.
-func (app *App) buildArtifact(ctx context.Context, def []byte, arch string, libraryRef string) (*build.BuildInfo, error) {
-	bi, err := app.buildClient.Submit(ctx, bytes.NewReader(def),
+func (app *App) buildArtifact(ctx context.Context, arch string, libraryRef string) (*build.BuildInfo, error) {
+	def, err := app.getBuildDef()
+	if err != nil {
+		return nil, fmt.Errorf("build definition error: %w", err)
+	}
+
+	// Validate def file
+	files, err := app.buildClient.ExtractFiles(ctx, bytes.NewReader(def))
+	if err != nil {
+		return nil, fmt.Errorf("def file validation error: %w", err)
+	}
```

Comment:
The error message references validation, when it's really trying to get a list of files. Maybe make that a bit clearer for the user:

```suggestion
	files, err := app.buildClient.ExtractFiles(ctx, bytes.NewReader(def))
	if err != nil {
		return nil, fmt.Errorf("failed to determine build context files: %w", err)
	}
```